hw12/Lesson12.py

```diff
@@ -94,16 +90,18 @@ def my_func():
     print('This is my func')
 
 
-my_function = my_decorator_func(my_func)
-
 my_func()
 
 
 class MyCustomException(Exception):
     def __init__(self):
         message = 'Custom exception is occurred!'
+        self.logging()
         super().__init__(message)
 
+    def logging(self):
+        return self
```

Comment:
write to file `self.message` of exception and `datetime` of it here in `logging` instead of this code block

> try:
>     raise MyCustomException()
> except MyCustomException as error_:
>     e = error_
>     now_ = datetime.datetime.now().strftime("%H:%M")
>     json_text = json.dumps(f'{e} at {now_}')
>     json_text = json.loads(str(json_text))
>     with open('Deco.json', 'w', encoding='UTF-8') as file:
>         json.dump(json_text, file)
> finally:
>     print('The error text was recorded in the Deco.json')